py/mpprint.c

```diff
@@ -353,6 +353,14 @@ int mp_print_float(const mp_print_t *print, mp_float_t f, char fmt, unsigned int
 
     char *s = buf;
 
+    if ((flags & PF_FLAG_ALWAYS_DECIMAL) && strchr(buf, '.') == NULL && strchr(buf, 'e') == NULL && strchr(buf, 'n') == NULL) {
+        if ((size_t)(len + 2) < sizeof(buf)) {
```

Comment:
Is it possible to remove this buffer size check by allocating 2 more bytes to `buf`, and then passing `sizeof(buf) - 2` to `mp_format_float()` above, thereby guaranteeing there will always be enough space to add these characters?

Same comment for the `PF_FLAG_ADD_PERCENT` check below, so add 3 bytes in total to `buf` and actually pass `sizeof(buf) - 3` to `mp_format_float()`.